### PR TITLE
Updated composer.json with new library versions

### PR DESCRIPTION
**Pull request by buskamuza**
### Description

Updated `composer.json` with library versions that support PHP 7.2. Magento behavior is not validated/fixed. The purpose of the PR is to upgrade dependent libraries, so the work on separate libraries can be unblocked.
The target branch is `libs-upgrade`, which is a temporary branch for libs upgrade work.

Includes https://github.com/magento-engcom/php-7.2-support/pull/30 (merged in).
Main changes are in https://github.com/magento-engcom/php-7.2-support/pull/41/files#diff-a787f6a8f1d79db8c096cfe917d4d07b

### Fixed Issues (if relevant)

1. magento/magento2#24: 3rd-party libraries do not support PHP 7.2

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/AdminNotification/composer.json [hunk 1]
```diff
@@ -13,6 +13,9 @@
         "magento/module-store": "100.3.*",
         "magento/module-ui": "100.3.*"
     },
+    "require-dev": {
+        "php": "7.0.2|7.0.4|~7.0.6|~7.1.0|~7.2.0"
```
Comment (YevSent) [inline app/code/Magento/AdminNotification/composer.json:17]: Why do we still need 7.0.x?

File: lib/internal/Magento/Framework/composer.json [hunk 2]
```diff
@@ -23,13 +23,13 @@
...
-        "colinmollenhour/php-redis-session-abstract": "~1.2.2",
-        "composer/composer": "1.4.1",
+        "colinmollenhour/php-redis-session-abstract": "~1.3.8",
+        "composer/composer": "~1.6.0",
         "magento/zendframework1": "~1.13.0",
         "monolog/monolog": "^1.17",
         "oyejorge/less.php": "~1.7.0",
-        "symfony/console": "~2.3, !=2.7.0",
-        "symfony/process": "~2.1",
+        "symfony/console": "~3.4.0",
```
Comment (YevSent) [inline lib/internal/Magento/Framework/composer.json:31]: Symfony console and process 4.0.4 have been released https://github.com/symfony/console/releases https://github.com/symfony/process/releases. Maybe, it's better to use them.

File: app/code/Magento/AdvancedPricingImportExport/composer.json [hunk 3]
```diff
@@ -15,6 +15,9 @@
         "magento/module-import-export": "100.3.*",
         "magento/module-store": "100.3.*"
     },
+    "require-dev": {
```
Comment (YevSent) [inline app/code/Magento/AdvancedPricingImportExport/composer.json:18]: Could you explain why `require-dev` section is needed?

File: lib/internal/Magento/Framework/composer.json (hunk 2 above)
Comment (buskamuza) [reply lib/internal/Magento/Framework/composer.json:31]: Some other libraries are incompatible with version 4.

UPD: looks like it's "phpcpd" only. Could update it to a newer version, and then symfony libs can also be updated to ver. 4.
Right now I have a problem with defining PHP 7.2.0 version in "require-dev" section - composer validation fail on 7.2.0 with `This package requires php 7.0.2|7.0.4|~7.0.6|~7.1.0 but your PHP version (7.2.0) does not satisfy that requirement.`. Will investigate this and then push the changes. This also causes update of PHPUnit (updating to the latest, v. 7).

File: app/code/Magento/AdminNotification/composer.json (hunk 1 above)
Comment (buskamuza) [reply app/code/Magento/AdminNotification/composer.json:17]: Left it there for now due to having our internal builds running on 7.0. If I remove it, the builds will fail. We should be able to remove it soon.

File: app/code/Magento/AdvancedPricingImportExport/composer.json (hunk 3 above)
Comment (buskamuza) [reply app/code/Magento/AdvancedPricingImportExport/composer.json:18]: In general - for development purposes only, not used in production.
In our case, we agreed to use this section as an indicator that PHP 7.2 support is experimental yet, and should not be used in production. But looks like Composer doesn't really like an idea of having the same dependency being declared in two places, so we might need to reconsider the solution.

**Comment by buskamuza** [comment]
https://github.com/magento-engcom/php-7.2-support/pull/86 covers the same

**Pull request closed without merging**